src/backend/distributed/master/master_delete_protocol.c

```diff
@@ -216,25 +219,24 @@ master_drop_all_shards(PG_FUNCTION_ARGS)
 
 
 /*
- * master_drop_sequences attempts to drop a list of sequences on a specified
- * node. The "IF EXISTS" clause is used to permit dropping sequences even if
- * they may not exist. Returns true on success, false on failure.
+ * master_drop_sequences attempts to drop a list of sequences on all nodes.
+ * The "IF EXISTS" clause is used to permit dropping sequences even if
+ * they may not exist.
  */
 Datum
 master_drop_sequences(PG_FUNCTION_ARGS)
 {
 	ArrayType *sequenceNamesArray = PG_GETARG_ARRAYTYPE_P(0);
-	text *nodeText = PG_GETARG_TEXT_P(1);
-	int64 nodePort = PG_GETARG_INT64(2);
-	bool dropSuccessful = false;
-	char *nodeName = TextDatumGetCString(nodeText);
-
 	ArrayIterator sequenceIterator = NULL;
 	Datum sequenceText = 0;
 	bool isNull = false;
-
 	StringInfo dropSeqCommand = makeStringInfo();
 
+	if (!EnableDDLPropagation)
```

Comment:
Also, I think this has some implications that would be the following:
```SQL
SELECT start_metadata_sync_to_node('localhost', 9700);
SELECT start_metadata_sync_to_node('localhost', 9701);
SELECT start_metadata_sync_to_node('localhost', 9702);
SET citus.shard_count TO 4;
SET citus.shard_replication_factor TO 1;

CREATE TABLE test (a int, b serial);
select create_distributed_table('test', 'a');
UPDATE pg_dist_partition SET repmodel='s' WHERE logicalrelid='test'::regclass;

insert into test (a) VALUES (1);
\c - - - 9700
insert into test (a) VALUES (2);
insert into test (a) VALUES (3);

\c - - - 5432
-- another time the user comes int and does the following
set citus.enable_ddl_propagation TO off;
DROP TABLE test;
CREATE TABLE test (a int, b serial);
select create_distributed_table('test', 'a');
UPDATE pg_dist_partition SET repmodel='s' WHERE logicalrelid='test'::regclass;

SELECT start_metadata_sync_to_node('localhost', 9700);
SELECT start_metadata_sync_to_node('localhost', 9701);
SELECT start_metadata_sync_to_node('localhost', 9702);

\c - - - 9700
-- we didn't drop the sequence thus continued
-- where seq_b is before DROP command
insert into test (a) VALUES (1);

SELECT * FROM test;
```

But, before testing that, the code fails as below:
```SQL
SELECT start_metadata_sync_to_node('localhost', 9700);
SELECT start_metadata_sync_to_node('localhost', 9701);
SELECT start_metadata_sync_to_node('localhost', 9702);
SET citus.shard_count TO 4;
SET citus.shard_replication_factor TO 1;

CREATE TABLE test (a int, b serial);
select create_distributed_table('test', 'a');
UPDATE pg_dist_partition SET repmodel='s' WHERE logicalrelid='test'::regclass;

insert into test (a) VALUES (1);
\c - - - 9700
insert into test (a) VALUES (2);
insert into test (a) VALUES (3);

\c - - - 5432
-- another time the user comes int and does the following
set citus.enable_ddl_propagation TO off;
DROP TABLE test;
CREATE TABLE test (a int, b serial);
select create_distributed_table('test', 'a');
UPDATE pg_dist_partition SET repmodel='s' WHERE logicalrelid='test'::regclass;
postgres=# SELECT start_metadata_sync_to_node('localhost', 9700);
server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
The connection to the server was lost. Attempting reset: Failed.
!> 

```
Could you check what's going on?